packages/ab-testing/src/index.ts

```diff
@@ -1,12 +1,25 @@
 import crc32 from 'fast-crc32c/impls/js_crc32c';
 
+type UserProfile = { [s: string]: string };
+
 type ForceInclude = {
     [key: string]: string[];
 };
 
-type Cohort = {
+type AllocationFields = { [key: string]: string[] };
+
+type AllocationV2 = {
+    range: [number, number][];
+    fields?: AllocationFields;
+};
+
+type AllocationV1 = [number, number][];
+
+type Allocation = AllocationV1 | AllocationV2;
+
+export type Cohort = {
     name: string;
-    allocation?: [number, number][];
+    allocation?: Allocation;
```

Comment:
how about

```ts
type Cohort = {
  name: string;
  allocation?: [number, number][];
  force_include?: ForceInclude;
  // or any other expressive names, like `requires`, `filters`
  matches?: ForceInclude;
}
```

benefits are 1) a flat data structure and 2) a simpler type interface.

Reply:
To me this is a good idea, the only caveat is that there is nothing that implicitly states that this belongs to allocation. Perhaps this approach but with allocation_matches as the key. @ramon-villain should we switch to this approach?

Reply:
yeah `allocation_matches` or `allocation_filters` looks good

Reply:
Yes… this also makes it backwards compatible by default… I think we can go with `allocation_criteria` or `allocation_matches`